src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -187,5 +173,35 @@ public static IServiceCollection AddSqlServerManagement<TVersion>(this IServiceC
 
             return services;
         }
+
+        /// <summary>
+        /// Adds an <see cref="SqlRetryLogicBaseProvider"/> to be used by SqlConnection and SqlCommand
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="NotImplementedException">When the retry mode is unkown</exception>
+        private static IServiceCollection AddSqlRetryLogicProvider(this IServiceCollection services)
+        {
+            services.TryAddSingleton<SqlRetryLogicBaseProvider>(p =>
+            {
+                SqlServerDataStoreConfiguration config = p.GetRequiredService<IOptions<SqlServerDataStoreConfiguration>>().Value;
+
+                // set default
+                if (config.Retry == null)
+                {
+                    config.Retry = new SqlClientRetryOptions();
```

Comment:
We can move this to `SqlServerDataStoreConfiguration.cs`, right? By default, the property can be populated

Reply:
And optionally if someone really wants it to be `null` (because they explicitly set it in their config), then we could treat it like `None`